src/coreclr/hosts/inc/coreclrhost.h

```diff
@@ -126,7 +126,7 @@ CORECLR_HOSTING_API(coreclr_execute_assembly,
 //
 // Callback types used by the hosts
 //
-typedef bool(CORECLR_CALLING_CONVENTION BundleProbeFn)(const char* path, int64_t* offset, int64_t* size);
+typedef bool(CORECLR_CALLING_CONVENTION BundleProbeFn)(const char* path, int64_t* offset, int64_t* sizes);
```

Comment:
Why can't we pass two separate parameters?

Reply:
I am concerned about framework-dependent scenarios where we may be calling the 5.0 version of the probe. Having the same signature allows the old clients to be forwards-compatible.
I.E. 5.0 probe will use only the first element of the array, which contains file size, which is compatible with the new `{size, uncompressedSize}`.

Passing an extra parameter "may work", depending on calling conventions. Passing an array does not need to rely on that.

Reply:
BTW - if we may need another piece of information from the probe in the future, we can continue adding optional array elements.

Reply:
But framework dependent app should use `hostpolicy.dll` which is next to the runtime - we should not be statically linking `hostpolicy` anywhere but superhost. So far we've made the assumption that `hostpolicy`/runtime interface is not "public" and can be change at any time without versioning considerations, because we always ship `hostpolicy` and runtime together.

Reply:
Hmm. I think I saw adding an extra parameter causing failures in framework-dependent 5.0 tests on OSX. 
Although we also had uninitialized data problem as well and it could be just that. Let me think about this. 

I thought we could have a case where 6.0 clr calls the probe and there is 5.0 hostpolicy on the other end. Perhaps this can't happen.

Reply:
I think you are correct. The probe is always in sync with runtime. `hostpolicy` is either bundled together with runtime in superhost or picked up next to the runtime. Either way it will match the version/build. 

That allows just using another parameter, so I have reverted the change that packs sizes in an array.

Reply:
I have reverted the change that packs sizes into an array. The failures in framework-dependent 5.0 tests on OSX were indeed caused by uninitialized data (although they looked like a probe mismatch)